keymanapp/keyman
# feat(web): `&newLayer` and `&oldLayer`

Fixes #6364.
Fixes #6112. (I think).

Removes `&layerChanged` and adds `&newLayer` and `&oldLayer` system stores with the additional nuances around values as described in #6364.

# User Testing

## Setup 

Android/IOS: Run these tests with the obolo_chwerty_6351 keyboard available in the Test-14.0 (Keyman - Windows (Desktop/Developer)) artifacts.

Web: The obolo_chwerty_6351 keyboard for these tests is available under **English.../obolo_chwerty_6351** in the globe menu.

1. touch **English...** first:

   ![image](https://user-images.githubusercontent.com/4498365/159345164-894e227b-f8c3-4c9b-9a98-0dff63fd4431.png)

2. Then **obolo_chwerty_6351** should be available in the submenu that appears:

   ![image](https://user-images.githubusercontent.com/4498365/159345087-b8df287a-4ab4-4fd8-879f-90f49c1f28b7.png)

## Base Configurations

These tests should be run on Android, iOS and Web.

**GROUP_ANDROID:** Run the Keyman app, either in Android emulator or real device. Version of Android is not important. Test the System keyboard and In-App keyboard.
**GROUP_IOS:** Run the Keyman app, either in iPhone simulator or real device. Version of iOS is not important. Test the System keyboard and In-App keyboard.
**GROUP_WEB:** Run testing/unminified, either in Chrome desktop, with mobile emulation, or mobile device/emulator. The obolo_chwerty_6351 keyboard for this test is available under English.../obolo_chwerty_6351 in the globe menu.

## Tests

* **TEST_STARTS_WITH_SHIFT:** Make sure that the initial shift layer is working correctly.
 
  1. Verify that the initial layer when the keyboard is activated in an empty text area is Shift.
  2. Press an alphabet key on the shift layer to verify that the output is correctly shifted.
  3. Verify that the layer switches to the base layer.
  4. Press another alphabet key, verify that the output is lower case.
  
* **TEST_CAPS:** Verify behaviour of Caps Lock

  1. Double-tap on the Shift key to activate Caps Lock.
  2. Verify that the next alphabet key pressed generates capitalised output.
  3. Verify that the layer remains the Caps Lock layer.
  4. Press the <kbd>CH</kbd> key and verify that it is also capitalised.
  5. Press Shift again to return to the base layer.
  6. Verify that the next alphabet key pressed generates lower case output.

# SUITE_SOS_IOS

* **TEST_6112:** Please retest the scenario in #6112.
